overhaul-literate/src/main/java/org/aya/concrete/remark2/AyaMdParser.java

```diff
@@ -0,0 +1,281 @@
+// Copyright (c) 2020-2022 Tesla (Yinsen) Zhang.
+// Use of this source code is governed by the MIT license that can be found in the LICENSE.md file.
+package org.aya.concrete.remark2;
+
+import kala.collection.Seq;
+import kala.collection.immutable.ImmutableSeq;
+import kala.collection.mutable.MutableList;
+import org.aya.concrete.GenericAyaParser;
+import org.aya.generic.util.InternalException;
+import org.aya.pretty.doc.Doc;
+import org.aya.pretty.doc.Style;
+import org.aya.util.error.SourceFile;
+import org.aya.util.error.SourcePos;
+import org.commonmark.node.*;
+import org.commonmark.parser.IncludeSourceSpans;
+import org.commonmark.parser.Parser;
+import org.jetbrains.annotations.Contract;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+public class AyaMdParser {
+  public static final char LINE_SEPARATOR = '\n';
+  private @NotNull
+  final String code;
+
+  /**
+   * For empty line that end with EOF, the index points to EOF<br/>
+   * For empty line that end with \n, the index points to \n
+   */
+  private @NotNull
+  final ImmutableSeq<Integer> linesIndex;
+  private @NotNull
+  final SourceFile file;
+
+  public AyaMdParser(@NotNull SourceFile file) {
+    this.file = file;
+    this.code = file.sourceCode()
+      .replace("\r\n", "\n")    // TODO: any elegant way?
+      .replace('\r', '\n');
```

Comment:
Try `TestRunner.trimCRLF`: https://github.com/aya-prover/aya-dev/blob/b1af157cee37999fa21ed22cf488945d77b0672e/base/src/test/java/org/aya/test/TestRunner.java#L162-L164

We can move this to `StringUtil`

Reply:
AMAZING!